internal/common/types.go

```diff
@@ -238,9 +266,10 @@ type ImageRequest struct {
 
 // ComposeRequest is used to submit a new compose to the store
 type ComposeRequest struct {
-	BlueprintName   string         `json:"blueprint_name"`
-	ComposeID       uuid.UUID      `json:"uuid"`
-	Distro          Distribution   `json:"distro"`
-	Arch            Architecture   `json:"arch"`
-	RequestedImages []ImageRequest `json:"requested_images"`
+	Blueprint       blueprint.Blueprint `json:"blueprint"`
+	ComposeID       uuid.UUID           `json:"uuid"`
+	Distro          Distribution        `json:"distro"`
+	Arch            Architecture        `json:"arch"`
+	Repositories    []rpmmd.RepoConfig  `json:"repositories"`
+	RequestedImages []ImageRequest      `json:"requested_images"`
```

Comment:
I'm basically still working on the structure. Previously I thought it would be enough to have the name, but since we want to move out the blueprint storage it has to take the blueprint as a structure. But sure, I'll try to explain it in a commit message.